source/client/factories_impl.cc

```diff
@@ -172,15 +174,19 @@ TerminationPredicateFactoryImpl::TerminationPredicateFactoryImpl(const Options&
 TerminationPredicatePtr
 TerminationPredicateFactoryImpl::create(Envoy::TimeSource& time_source, Envoy::Stats::Scope& scope,
                                         const Envoy::MonotonicTime scheduled_starting_time) const {
-  TerminationPredicatePtr duration_predicate = std::make_unique<DurationTerminationPredicateImpl>(
-      time_source, options_.duration(), scheduled_starting_time);
-  TerminationPredicate* current_predicate = duration_predicate.get();
+  TerminationPredicatePtr root_predicate = std::make_unique<NullTerminationPredicateImpl>();
+  if (options_.duration() > 0s) {
```

Comment:
This may be a bit surprising to unaware users. Especially if zero is the default.

Additionally what would happen in the gRPC world if we send a request to the NH server with an infinite duration and how will the client terminate such a run and get the results? The same question likely applies to local runs - how do we terminate and get results?

Do we want to do some or all of the following?
a) print out on the screen + log saying that this is a run with an infinite duration and an instruction of how to terminate it.
b) should we use -1 instead of zero to avoid the "runs infinitely by default" situation? It would also help to make this change backwards compatible.
c) should we improve the documentation of the option indicating how does one terminate an infinite run and get results?

Reply:
> This may be a bit surprising to unaware users. Especially if zero is the default.
> 

Our default duration actually is 5 seconds, but that gets defined elsewhere in the options implementation header. That part doesn't change in this PR.

> Additionally what would happen in the gRPC world if we send a request to the NH server with an infinite duration and how will the client terminate such a run and get the results? The same question likely applies to local runs - how do we terminate and get results?
> 

For local runs, there's no way to terminate (gracefully) today. But this PR purposely didn't worry about that, as the feature was requested in a context where [results are irrelevant](https://github.com/envoyproxy/nighthawk/issues/333#issuecomment-645011251). So this PR helps by proving a way to just generate infinite load. 
The gRPC service would behave identical. 

However, I did have [similar thoughts](https://github.com/envoyproxy/nighthawk/issues/333#issuecomment-644989540) before starting on this, so I also [added a follow up](https://github.com/envoyproxy/nighthawk/pull/367) to that teaches the CLI about handling signals and wires through a way for `ProcessImpl` to quickly and gracefully exit and yield results. `RemoteProcessImpl` still has a `TODO` for that, but the path I had in mind there would send a gRPC message to request cancellation of the currently active `ProcessImpl` over there, which would then be propagated to the active `ProcessImpl` running within this service -- if any (we can only have a single one active at a time today). 

> Do we want to do some or all of the following?
> a) print out on the screen + log saying that this is a run with an infinite duration and an instruction of how to terminate it.

Right now the CLI will print that execution will target 0 seconds via an informational log line. I will make that print out "infinite" instead.

> b) should we use -1 instead of zero to avoid the "runs infinitely by default" situation? It would also help to make this change backwards compatible.

Alternatively, we could also try to add an explicit `--no-duration` flag, and make this value an internal detail. If we back out the current change, zero would be suitable for that, as we guard against that value being set by the user. I can go either way.

> c) should we improve the documentation of the option indicating how does one terminate an infinite run and get results?
That probably belongs in the follow up I mentioned above?